src/main/java/org/opensearch/knn/training/VectorReader.java

```diff
@@ -225,5 +220,32 @@ public void onFailure(Exception e) {
                 listener.onFailure(e);
             }
         }
+
+        /**
+         * Extracts vectors from the hits in a search response
+         *
+         * @param searchResponse Search response to extract vectors from
+         * @param vectorsToAdd number of vectors to extract
+         * @return list of vectors
+         */
+        private List<Float[]> extractVectorsFromHits(SearchResponse searchResponse, int vectorsToAdd) {
+            SearchHit[] hits = searchResponse.getHits().getHits();
+            List<Float[]> trainingData = new ArrayList<>();
+            for (int vector = 0; vector < vectorsToAdd; vector++) {
+                Map<String, Object> currentMap = hits[vector].getSourceAsMap();
+                // The field name may be a nested field, so we need to split it and traverse the map.
+                // Example fieldName: "my_field" or "my_field.nested_field.nested_nested_field"
+                String[] fieldPath = fieldName.split("\\.");
+
+                for (int pathPart = 0; pathPart < fieldPath.length - 1; pathPart++) {
+                    currentMap = (Map<String, Object>) currentMap.get(fieldPath[pathPart]);
```

Comment:
we should check if currentMap.get(fieldPath[pathPart]) != null and then only type cast the value otherwise there is a possibility of NullPointerException.